src/core/server/base.js

```diff
@@ -109,11 +122,11 @@ function baseServer(routes, createStore, { appInstanceName = appName } = {}) {
 
       if (err) {
         log.error({ err, req });
-        return res.status(500).end(errorString);
+        return showErrorPage(res, 500);
       }
 
       if (!renderProps) {
-        return res.status(404).end('Not found.');
+        return showErrorPage(res, 404);
```

Comment:
I think we're going to want these to render the full application at some point. That could be a separate issue to keep this smaller if you'd like.